internal/security/bootstrapper/command/setupacl/aclpolicies.go

```diff
@@ -184,6 +189,53 @@ func (c *cmd) getOrCreateRegistryPolicy(tokenID, policyName, policyRules string)
 
 // getPolicyByName gets policy by policy name, returns nil if not found
 func (c *cmd) getPolicyByName(tokenID, policyName string) (*Policy, error) {
+	var isExistingPolicy bool
+	isExistingPolicy = false
+
+	policyListURL, err := c.getRegistryApiUrl(consulPolicyListAPI)
+	if err != nil {
+		return nil, err
+	}
+
+	policyListReq, err := http.NewRequest(http.MethodGet, policyListURL, http.NoBody)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to prepare policyListReq request for http URL %s: %w", policyListURL, err)
+	}
+
+	policyListReq.Header.Add(share.ConsulTokenHeader, tokenID)
+	policyListResp, err := c.client.Do(policyListReq)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to GET policy list request for http URL %s: %w", policyListURL, err)
+	}
+	defer policyListResp.Body.Close()
+
+	var policyList PolicyListResponse
+
+	decode_err := json.NewDecoder(policyListResp.Body).Decode(&policyList)
+	if decode_err != nil {
+		return nil, fmt.Errorf("Failed to decode policy list reponse: %w", decode_err)
+	}
+
+	switch policyListResp.StatusCode {
+	case http.StatusOK:
+		for _, policy := range policyList {
+			if policyName == policy.Name {
+				isExistingPolicy = true
+				break
+			}
+		}
```

Comment:
If we already found policy here, we can just return policy? why we need to find policy by name again below? (IMO, kind of redundant logic below)

Reply:
No, it's not.  The list API only returns the names of the policies.  Still need another call to actually get the policy (i.e. it is just chaining off to the original code).

Reply:
that is not true: see this: https://www.consul.io/api-docs/acl/policies#list-policies

Reply:
@jim-wang-intel   I understand, I think, what you were saying.  We only really need the name and the ID, not the text of the policy.  I think I like the solution as presented because it addresses the bug as stated (error in consul log due to get of non-existent policy.)  Since this is bootstrapping logic, it is not performance critical that we get the answer we need in one or two REST API calls.  I think we should implement the simplest and most readable fix.  We can refactor later if we want.

Reply:
yep, it sounds ok